docs/design_guide.md

```diff
@@ -0,0 +1,285 @@
+- DO: Place 1 space on each side of arithmetic and assignment operators
+- DO NOT: Place more than 1 space before or after assignments operators, i.e. do not align `=` vertically when assigning variable values
+- DO NOT: Place spaces around the `=` sign when using with keyword arguments to a function
+- DO NOT: Leave trailing white spaces at the end of lines or in blank lines
+
+
+**Quotes**
+
+Use double quotes: `"your string"` for strings, `"""comment"""` for comments.
+
+
+### Documentation
+
+*Note: this only covers how documentation within code should be formatted, for information on how documentation for the library will be developed see [below](#library-documentation).*
+
+**Comments** should be complete sentences used to elaborate on the code.
+
+- Block comments (1 or more lines each beginning with a `# `) should be indented to match the code that *follows* them.
+- Inline Comments (a comment on the same line as the code) should be at least 2 spaces away from the statement and server a purpose. They should disambiguate the line of code, either in function or intended purpose.
+- For comments aimed at developers try to use `TODO`, `FIXME`, `NOTE`, etc. as much as possible.
+
+
+**Documentation Strings** should follow the [Google Style](http://google.github.io/styleguide/pyguide.html) guidelines and be compatible with [Sphinx](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html). Below are some examples of what different docstrings *might* look like.
+
+**Class**
+```python
+"""One line class description.
+
+Longer explanation...
+...
+
+Attributes:
+    attr1: I am the first attribute for this class.
+"""
+```
+
+**Method**
+```python
+"""One line function description.
+
+Longer explanation...
+...
+
+Args
+    arg1 (int): I am an integer argument.
+    arg2 (str): I am a string argument.
+    arg3: I am argument without a type specified.
+
+Returns:
+    bool: This function returns a boolean.
+
+Raises:
+    TypeError: explanation of error.
+"""
+```
+
+### Naming
+
+| Name Type | Convention | Notes |
+| :---      | :---       | :---  |
+|Package, Module | `shortlowcase` | Underscores can be used for readability |
+|Class | `CapWords` | |
+| Type Variable | `CapWords_suffix`| |
+| Global Variables | `low_case_sep`| |
+| Functions | `low_case_sep`| |
+| Function Variables | `low_case_sep`| |
+| Function Arguments | `low_case_sep` | Use underscores sparingly |
+| Constants | `ALL_CAPS` | |
+
+## Development Environment
+
+The library will be written in Python 3.7, so use this version when developing. To see what version you currently have installed, run `python3 --version`. If the output is `Python 3.7.x`, you are good to go! Otherwise, [install](https://www.python.org/downloads/).
+
+Code can be linted using [flake8](https://flake8.pycqa.org/en/latest/index.html), just ensure it is installed for the correct version of Python (i.e. 3.7).
```

Comment:
This would be a great place to document how to set up the dev env. Perhaps it is just a `pip` command line referencing our requirements files?